src/main/java/net/apnic/rdap/error/controller/ErrorController.java

```diff
@@ -0,0 +1,190 @@
+    {
+        return createErrorResponse(status, null, context);
+    }
+
+    /**
+     * Constructs a new RDAPError object for the provided HttpStatus
+     * description, and context.
+     *
+     * @param status HttpStatus for the RDAPError
+     * @param description Description for the error object
+     * @param context The context the error request was made in. This will be
+     *                the request URL in most cases
+     */
+    private ResponseEntity<RDAPError> createErrorResponse(HttpStatus status,
+                                                         String description,
+                                                         String context)
+    {
+        RDAPError error =
+            rdapObjectFactory.createRDAPObject(RDAPError.class, context)
+                .setErrorCode(status.value())
+                .setTitle(status.getReasonPhrase());
+
+        if(description != null)
+        {
+            error.addDescription(description);
+        }
+
+        return new ResponseEntity<RDAPError>(error, responseHeaders,
+                                             status);
+    }
+
+    /**
+     * Spring error route handler.
+     *
+     * Function attempts to work out the cause of the error and provide a
+     * correct RDAPError resonse. If no specific cause can be worked out then
+     * a default internal server error is returned.
+     */
+    @RequestMapping(value="${error.path:/error}")
+    public ResponseEntity<RDAPError> error(HttpServletRequest request)
+    {
+        Throwable topCause = (Throwable)request.getAttribute(EXCEPTION_ATTRIBUTE);
+        String context = (String)request.getAttribute(ORIGIN_ATTRIBUTE);
+
+        if(topCause == null)
+        {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        System.out.println(topCause.getCause());
+        if(causedBy(topCause, MalformedRequestException.class))
+        {
+            return createErrorResponse(HttpStatus.BAD_REQUEST,
+                "Request could not be understood, malformed syntax", context);
+        }
+        else if(causedBy(topCause, ResourceNotFoundException.class))
+        {
+            return createErrorResponse(HttpStatus.NOT_FOUND, context);
+        }
+        return createErrorResponse(HttpStatus.INTERNAL_SERVER_ERROR, context);
+    }
+
+    /**
+     * Provides the configured error path for Spring.
+     */
+    @Override
+    public String getErrorPath()
+    {
+        return errorPath;
+    }
+
+    /**
+     * Generic spring handler for all exceptions that occur outside of Zuul
+     */
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<RDAPError> handleException(HttpServletRequest request)
+    {
+        String context = request.getRequestURL().toString();
+        return createErrorResponse(HttpStatus.INTERNAL_SERVER_ERROR, context);
+    }
+
+    /**
+     * Handles all Spring exceptions outside of Zuul where no rest handler can
```

Comment:
This would break the error handling that spring relies on as nothing would be able to get down to Spring and the paths that it registers. Also slows the server down as Zuul is generating much more expensive errors for things it does not care about.